coralproject/cay
# Form config refactor

## What does this PR do?

Before, there was one action `updateForm` that updated all parts of the form object in state. Unfortunately, our schema has nested properties, and they were being wiped on updates due to the way `Object.assign` works in javascript.

Now there are separate actions for each nested property. e.g. `updateFormHeader`, `updateFormFooter` etc. A better solution to this problem would be to flatten the form schema, or at least have everything in one `config` object that gets sent to the server. This would require a refactor of Elkhorn, and I kinda just want to get this working right now. So, this PR is a monkey-patch before a real refactor.
## How do I test this PR?
- update headers, footers, titles and other meta stuff about a form.
- change inactive message and make sure status doesn't change to closed
- `npm test`

@coralproject/frontend
